src/re/multi_grid/correlated_field.py

```diff
@@ -0,0 +1,119 @@
+from dataclasses import field
+from typing import Union
+
+import jax.numpy as jnp
+
+from ..model import Model
+from ..prior import NormalPrior
+from ..tree_math import ShapeWithDtype
+from .grid import Grid
+from .kernel import ICRKernel, Kernel, apply_kernel
+
+
+class ICRCorrelatedField(Model):
+    grid: Grid
+    kernel: Kernel
+    covariance: Union[Model, callable] = field(metadata=dict(static=False))
+    offset: Model = field(metadata=dict(static=False))
+    compress: bool
```

Comment:
Union[bool, dict]

Reply:
The attribute `compress` is not the same as the input parameter `compress`. It is set to true when compression is enabled (i.e. it is set to `self.compress = isinstance(compress, dict) and len(compress) > 0`). The line of thought for this choice is that we only need to know whether the kernel was compressed but we don't need the specific arguments anymore for successive re-compression after the first full compression was carried out.